internal/state/offset.go

```diff
@@ -0,0 +1,212 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package state
+
+import (
+	"bytes"
+	"encoding"
+	"errors"
+	"log"
+	"math"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	// defaultSaveInterval is the default duration between state file saves
+	defaultSaveInterval = 100 * time.Millisecond
+	// defaultQueueSize is the default capacity of the offset queue
+	defaultQueueSize = 2000
```

Comment:
do we need to keep this many in the queue? Is the purpose to reduce the likelihood of older items being dropped when the queue becomes full?